### PR TITLE
Arc WinitWindow and extract to render world

### LINKED ISSUE #11150: XError `BadWindow (invalid Window parameter)` when closing windows at seemingly random
## Bevy version
0.12.1

## Relevant system information
OS: Pop!_OS 22.04 (6.6.6-76060606-generic)
```
AdapterInfo { name: "llvmpipe (LLVM 15.0.7, 256 bits)", vendor: 65541, device: 0, device_type: Cpu, driver: "llvmpipe", driver_info: "Mesa 23.3.0-1pop0~1702935939~22.04~67e417a (LLVM 15.0.7)", backend: Vulkan }
```
## What you did
I was playing around with multiple windows, essentially like so:
```rs
use bevy::prelude::*;

#[derive(Resource)]
struct WindowCounter(pub usize);

fn main() {
    App::new()
        .insert_resource(WindowCounter(1))
        .add_plugins(DefaultPlugins.set(WindowPlugin {
            primary_window: Some(Window {
                title: "Window #0".into(),
                ..Default::default()
            }),
            ..Default::default()
        }))
        .add_systems(Update, open_new_window_on_keypress)
        .run();
}

fn open_new_window_on_keypress(
    input: Res<Input<KeyCode>>,
    mut commands: Commands,
    mut windows: ResMut<WindowCounter>,
) {
    if input.just_released(KeyCode::Space) {
        let name = format!("Window #{}", windows.0);

        commands.spawn((
            Window {
                title: name.clone(),
                ..Default::default()
            },
            Name::new(name),
        ));

        windows.0 += 1;
    }
}

```

## What went wrong
Upon closing any window (via Super + Q/Alt + F4 or the close button), there's a pretty small chance that the entire application crashes:
```ignore
2023-12-30T19:14:40.903590Z  INFO bevy_winit::system: Creating new window "Window #2" (1v1)
2023-12-30T19:14:40.903690Z  INFO log: Guessed window scale factor: 1    
2023-12-30T19:14:41.366602Z  INFO bevy_winit::system: Closing window 1v1
2023-12-30T19:14:41.367636Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event Focused(true)
2023-12-30T19:14:41.367732Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event ModifiersChanged(LOGO)
2023-12-30T19:14:41.367743Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event CursorMoved { device_id: DeviceId(X(DeviceId(2))), position: PhysicalPosition { x: 933.0, y: 202.0 }, modifiers: LOGO }
2023-12-30T19:14:41.368243Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 16, state: Pressed, virtual_keycode: Some(Q), modifiers: LOGO }, is_synthetic: true }
2023-12-30T19:14:41.368288Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event KeyboardInput { device_id: DeviceId(X(DeviceId(3))), input: KeyboardInput { scancode: 125, state: Pressed, virtual_keycode: None, modifiers: LOGO }, is_synthetic: true }
2023-12-30T19:14:41.369414Z  WARN bevy_winit: Window 1v1 is missing `Window` component, skipping event Destroyed
2023-12-30T19:14:41.927830Z  INFO bevy_winit::system: Creating new window "Window #3" (1v2)
2023-12-30T19:14:41.927959Z  INFO log: Guessed window scale factor: 1    
2023-12-30T19:14:42.278620Z  INFO bevy_winit::system: Closing window 1v2
2023-12-30T19:14:42.289309Z ERROR log: X11 error: XError {
    description: "BadWindow (invalid Window parameter)",
    error_code: 3,
    request_code: 148,
    minor_code: 1,
}    
thread 'main' panicked at /home/eats/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.28.7/src/platform_impl/linux/x11/event_processor.rs:936:30:
Failed to focus input context: XError { description: "BadWindow (invalid Window parameter)", error_code: 3, request_code: 148, minor_code: 1 }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Additional information
I'm not familiar enough with windowing in general, `winit` or how `bevy_winit` uses it to track down where exactly the error originates from, but I ran a handful of the `winit` examples I deemed relevant and didn't encounter any issues there, so I'm bringing it up here for now.

### LINKED ISSUE #11236: Multiple windows example panics when closing one of the windows
* Bevy version: 425570aa752b32ef26f6573b385d49d6254bd868 (main as of today)
* Linux/X11

## What you did

* Run the `multiple_windows` example with `cargo run --example multiple_windows`.

## What went wrong

Closing a window results in a panic with the following error:
```
bevy_winit: winit event loop returned an error: os error at
cargo/registry/src/winit-0.29.9/src/platform_impl/linux/x11/mod.rs:433:
Xlib error: X error: BadWindow (invalid Window parameter) (code: 3, request code: 148, minor code: 1)
```

## Additional information

Probably related to #10702

### LINKED ISSUE #11734: 2D examples crash on exit with "Couldn't get swap chain texture"
## Bevy version

main, commit 2a1ebc4ac

## \[Optional\] Relevant system information

```ignore
2024-02-06T02:02:22.879898Z  INFO bevy_render::renderer: AdapterInfo { name: "AMD Radeon Graphics (RADV GFX1103_R1)", vendor: 4098, device: 5567, device_type: IntegratedGpu, driver: "radv", driver_info: "Mesa 23.3.3", backend: Vulkan }
2024-02-06T02:02:23.344221Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Linux 39 Fedora Linux", kernel: "6.6.13-200.fc39.x86_64", cpu: "", core_count: "8", memory: "58.5 GiB" }
```

## What you did

`cargo run --example sprite`

## What went wrong

Everything runs fine until I close the window, then:

```ignore
thread 'Compute Task Pool (2)' panicked at crates/bevy_render/src/view/window/mod.rs:372:21:
Couldn't get swap chain texture, operation unrecoverable: The swap chain has been lost and needs to be recreated
stack backtrace:
   0: rust_begin_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:72:14
   2: bevy_render::view::window::prepare_windows
             at ./crates/bevy_render/src/view/window/mod.rs:372:21
   3: core::ops::function::FnMut::call_mut
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:166:5
   4: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:294:13
   5: <Func as bevy_ecs::system::function_system::SystemParamFunction<fn(F0,F1,F2,F3,F4,F5,F6,F7,F8) .> Out>>::run::call_inner
             at ./crates/bevy_ecs/src/system/function_system.rs:661:21
   6: <Func as bevy_ecs::system::function_system::SystemParamFunction<fn(F0,F1,F2,F3,F4,F5,F6,F7,F8) .> Out>>::run
             at ./crates/bevy_ecs/src/system/function_system.rs:664:17
   7: <bevy_ecs::system::function_system::FunctionSystem<Marker,F> as bevy_ecs::system::system::System>::run_unsafe
             at ./crates/bevy_ecs/src/system/function_system.rs:504:19
   8: bevy_ecs::schedule::executor::multi_threaded::MultiThreadedExecutor::spawn_system_task::{{closure}}::{{closure}}
             at ./crates/bevy_ecs/src/schedule/executor/multi_threaded.rs:535:26
   9: core::ops::function::FnOnce::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
  10: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
  11: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  12: __rust_try
  13: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  14: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  15: bevy_ecs::schedule::executor::multi_threaded::MultiThreadedExecutor::spawn_system_task::{{closure}}
             at ./crates/bevy_ecs/src/schedule/executor/multi_threaded.rs:530:23
  16: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::future::future::Future>::poll
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:297:9
  17: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:42
  18: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
  19: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  20: __rust_try
  21: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  22: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  23: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:9
  24: async_executor::Executor::spawn::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:158:20
  25: async_task::raw::RawTask<F,T,S,M>::run::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/raw.rs:550:21
  26: core::ops::function::FnOnce::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
  27: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
  28: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  29: __rust_try
  30: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  31: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  32: async_task::raw::RawTask<F,T,S,M>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/raw.rs:549:23
  33: async_task::runnable::Runnable<M>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/runnable.rs:781:18
  34: async_executor::Executor::run::{{closure}}::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:254:21
  35: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:449:33
  36: async_executor::Executor::run::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:261:32
  37: futures_lite::future::block_on::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:99:19
  38: std::thread::local::LocalKey<T>::try_with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:270:16
  39: std::thread::local::LocalKey<T>::with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:246:9
  40: futures_lite::future::block_on
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:78:5
  41: bevy_tasks::task_pool::TaskPool::new_internal::{{closure}}::{{closure}}::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:180:37
  42: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  43: __rust_try
  44: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  45: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  46: bevy_tasks::task_pool::TaskPool::new_internal::{{closure}}::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:174:43
  47: std::thread::local::LocalKey<T>::try_with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:270:16
  48: std::thread::local::LocalKey<T>::with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:246:9
  49: bevy_tasks::task_pool::TaskPool::new_internal::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:167:25
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Encountered a panic in system `bevy_render::view::window::prepare_windows`!
thread 'main' panicked at crates/bevy_render/src/pipelined_rendering.rs:145:45:
called `Result::unwrap()` on an `Err` value: RecvError
stack backtrace:
   0: rust_begin_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/result.rs:1653:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/result.rs:1077:23
   4: bevy_render::pipelined_rendering::update_rendering::{{closure}}::{{closure}}::{{closure}}
             at ./crates/bevy_render/src/pipelined_rendering.rs:145:21
   5: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::future::future::Future>::poll
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:297:9
   6: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:42
   7: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
   8: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
   9: __rust_try
  10: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  11: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  12: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:9
  13: async_executor::Executor::spawn::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:158:20
  14: async_task::raw::RawTask<F,T,S,M>::run::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/raw.rs:550:21
  15: core::ops::function::FnOnce::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
  16: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
  17: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  18: __rust_try
  19: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  20: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  21: async_task::raw::RawTask<F,T,S,M>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/raw.rs:549:23
  22: async_task::runnable::Runnable<M>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-task-4.7.0/src/runnable.rs:781:18
  23: async_executor::Executor::run::{{closure}}::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:254:21
  24: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:449:33
  25: async_executor::Executor::run::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-executor-1.8.0/src/lib.rs:261:32
  26: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::future::future::Future>::poll
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:297:9
  27: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:42
  28: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/panic/unwind_safe.rs:272:9
  29: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  30: __rust_try
  31: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  32: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  33: <futures_lite::future::CatchUnwind<F> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:588:9
  34: bevy_tasks::task_pool::TaskPool::execute_global_scope::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:502:22
  35: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:446:33
  36: bevy_tasks::task_pool::TaskPool::execute_global_scope::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:506:41
  37: bevy_tasks::task_pool::TaskPool::scope_with_executor_inner::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:433:89
  38: futures_lite::future::block_on::{{closure}}
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:99:19
  39: std::thread::local::LocalKey<T>::try_with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:270:16
  40: std::thread::local::LocalKey<T>::with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:246:9
  41: futures_lite::future::block_on
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-lite-2.2.0/src/future.rs:78:5
  42: bevy_tasks::task_pool::TaskPool::scope_with_executor_inner
             at ./crates/bevy_tasks/src/task_pool.rs:389:13
  43: bevy_tasks::task_pool::TaskPool::scope_with_executor::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:320:17
  44: std::thread::local::LocalKey<T>::try_with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:270:16
  45: std::thread::local::LocalKey<T>::with
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/thread/local.rs:246:9
  46: bevy_tasks::task_pool::TaskPool::scope_with_executor
             at ./crates/bevy_tasks/src/task_pool.rs:316:9
  47: bevy_render::pipelined_rendering::update_rendering::{{closure}}
             at ./crates/bevy_render/src/pipelined_rendering.rs:141:30
  48: bevy_ecs::world::World::resource_scope
             at ./crates/bevy_ecs/src/world/mod.rs:1672:22
  49: bevy_render::pipelined_rendering::update_rendering
             at ./crates/bevy_render/src/pipelined_rendering.rs:138:5
  50: core::ops::function::Fn::call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:79:5
  51: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/alloc/src/boxed.rs:2021:9
  52: bevy_app::app::SubApp::extract
             at ./crates/bevy_app/src/app.rs:170:9
  53: bevy_app::app::App::update
             at ./crates/bevy_app/src/app.rs:268:13
  54: bevy_winit::run_app_update_if_should
             at ./crates/bevy_winit/src/lib.rs:717:9
  55: bevy_winit::handle_winit_event
             at ./crates/bevy_winit/src/lib.rs:597:21
  56: bevy_winit::winit_runner::{{closure}}
             at ./crates/bevy_winit/src/lib.rs:281:9
  57: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:294:13
  58: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:294:13
  59: winit::platform_impl::platform::x11::EventLoop<T>::single_iteration
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/x11/mod.rs:608:17
  60: winit::platform_impl::platform::x11::EventLoop<T>::poll_events_with_timeout
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/x11/mod.rs:547:9
  61: winit::platform_impl::platform::x11::EventLoop<T>::pump_events
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/x11/mod.rs:453:13
  62: winit::platform_impl::platform::x11::EventLoop<T>::run_on_demand
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/x11/mod.rs:414:19
  63: winit::platform_impl::platform::EventLoop<T>::run_on_demand
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/mod.rs:821:56
  64: winit::platform_impl::platform::EventLoop<T>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/platform_impl/linux/mod.rs:814:9
  65: winit::event_loop::EventLoop<T>::run
             at /home/alercah/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.10/src/event_loop.rs:249:9
  66: bevy_winit::winit_runner
             at ./crates/bevy_winit/src/lib.rs:296:23
  67: core::ops::function::FnOnce::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
  68: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
  69: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/alloc/src/boxed.rs:2007:9
  70: bevy_app::app::App::run
             at ./crates/bevy_app/src/app.rs:308:9
  71: sprite::main
             at ./examples/2d/sprite.rs:6:5
  72: core::ops::function::FnOnce::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

### PR DESCRIPTION
**Pull request by s-puig**
# Objective

Fixes #11236
~~(likely)~~ Fixes #11150 
~~(likely)~~ Fixes #11734

## Problem

Window crashes when using pipelined rendering under GPU-bound scenarios.

It is not exclusive to XWayland and is based on a race condition with pipelined rendering. 

Winit windows might be deleted on app world while the window is still alive in render world, if the render world window tries to access a winit resource (i.e. a surface or raw handle) it panics because the winit window no longer exists.
All those crashes go away by disabling pipelined rendering.

## Solution

- `Arc<WinitWindow>` and extract it when using pipelined rendering.

## Alternative solutions

- Delay deleting WinitWindow by a frame.

### CODE REVIEW

**Comment by Friz64** [comment]
I can confirm this fixes #11150 / #11734 for me.

**Review by james7132 — COMMENTED** [review]
Would it be possible to put this behind a feature flag that is enabled when "pipelined_rendering" is enabled on the top level Bevy crate? Or is this an issue when pipelined_rendering is disabled?

**Review by alice-i-cecile — APPROVED** [review]
@james7132 I don't agree with feature-flagging this :)

It looks like it should always work, even when pipelined rendering is off, and maintaining compile-gated code like this is substantially harder.

That said, I don't feel strongly enough to block on this point: the cost is small and the fix is important.

**Comment by JMS55** [comment]
Maybe we can wrap the window in an Arc or something?

**Comment by james7132** [comment]
My biggest concern here is that the extra frame that we're now running potentially has extra user or plugin logic running through it, which may run contrary to the expectations that closing the window immediately send an AppExit event.

With that said, the problem persists even with the cfg block, which comes with more complexity. I'm fine with this as is for now, but @JMS55's suggestion to internally wrap it in an Arc may be the right strategy here.

File: crates/bevy_render/src/view/window/mod.rs [hunk 1]
```diff
@@ -34,6 +36,20 @@ pub struct WindowRenderPlugin;
...
+        ))]
+        {
+            // TODO: This is added despite not checking if PipelinedRenderingPlugin is added or not,
+            // but we can't check right now (since it's added after this plugin)
+            if app.is_plugin_added::<bevy_winit::WinitPlugin>() {
+                app.add_plugins(winit::WinitWindowRenderPlugin);
+            } else {
+                bevy_utils::tracing::warn!("Winit feature was enabled but couldn't detect WinitPlugin. Are you sure you loaded this after WinitPlugin?");
+            }
+        }
```
Comment (james7132) [inline crates/bevy_render/src/view/window/mod.rs:52]: This potentially hard-couples bevy_render and bevy_winit. I'm not sure if we want to do this.

Comment (Friz64) [reply crates/bevy_render/src/view/window/mod.rs:52]: Take a look at this: https://github.com/Friz64/bevy/commit/0561754959e8051acf448a64a7ac982177a44def

It also works as a fix, and avoids the hard-coupling to `bevy_winit`. I'm not convinced that dynamic dispatch is the most elegant solution here, but it works.

**Pull request closed without merging**